core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java

```diff
@@ -296,7 +296,20 @@ public static InetSocketAddress getBindAddress(ServiceType service) {
   }
 
   /**
-   * Gets a local host name for the host this JVM is running on.
+   * Gets the local hostname to be used by the client. If this isn't configured, a non-loopback
+   * local hostname will be looked up.
+   *
+   * @return the local hostname for the client
+   */
+  public static String getClientHostName() {
+    if (Configuration.containsKey(PropertyKey.USER_HOSTNAME)) {
+      return Configuration.get(PropertyKey.USER_HOSTNAME);
+    }
+    return getLocalHostName();
```

Comment:
pretty much all hostname lookups will fall back on `getLocalHostName()` if they fail to find a specified configuration property